existing-solutions.tex

```diff
@@ -30,44 +30,45 @@ \subsection{Proof of Stake}
 not imply any noticeable energetic overhead and therefore is much more friendly for the
 environment. It also doesn't require users to have powerful computers to be able to
-have some involvement in the decision making.
+have some involvement in decision making.
 
 However, it comes with some serious issues. First of all, there is the infamous "nothing at stake"
 \cite{pos_flaws_nothing}
-problem which exploits the lack of any cost of the actual mining. In this case
+problem, which exploits the lack of any cost of the actual mining. In this case,
 there is no downside to staking several branches simultaneously in case of a fork.
-The next thing is the source of random entropy – as it needs to be deterministic
-and distributed it usually comes from the chain itself. This
-makes all of the elections completely predictable and leads to a strategy known
-commonly as "stake grinding", where the dishonest leader tries to rearrange the
-transactions to influence the result of the upcoming election. Next issue is the
-"long range attack"\cite{pos_flaws_long}.
-In the very beginning the stake is scattered among a small
+
+Ensuring that the source of entropy is distributed along the chain itself,
+makes all of the elections entirely deterministic and predictable and leads to a strategy
+known commonly as "stake grinding," where the dishonest leader tries to rearrange the
+transactions to influence the result of the upcoming election.
+
+Next issue is the "long range attack"\cite{pos_flaws_long}.
+In the very beginning, the stake is scattered among a small
 group of delegates that together have full control over the chain. After
-some time they can agree and start a concurrent chain from the same genesis
+some time, they can agree and start a concurrent chain from the same genesis
 and go ahead of the main one. This could lead to nasty frauds and would
 destabilize the trust over the chain.
 
-On the other hand, there are multiple approaches to deal with these problems. For instance
-the CASPER protocol introduces a "wrong voting penalty" which punishes the
+On the other hand, there are multiple approaches to deal with these problems. For instance,
+the CASPER protocol introduces a "wrong voting penalty," which punishes the
 voters that support conflicting forks to deal with nothing at stake\cite{casper}.
 However, there is a PoW mechanism in the background anyway. Another way is to use
 proof of fraud mechanisms known from BitcoinNG\cite{bcng},
 but they are hopeless in a situation where the chain forks over a transaction
-transferring stake between two accounts and the penalties are not always
+transferring stake between two accounts, and the penalties are not always
 sufficient. Also, it simply doesn't work until the
-conflicting branch is published. NXT deals with long range attack by forcefully
+conflicting branch is published. NXT deals with long-range attack by forcefully
```

Comment:
this "-" is unnecessary imo

Reply:
It seems like long-range attacks are written this way quite commonly: https://www.researchgate.net/publication/331313599_A_Survey_on_Long-Range_Attacks_for_Proof_of_Stake_Protocols for example. Wdyt?